### ISSUE TITLE
Performance: getConfigurableAttributes cache is broken

### ISSUE THREAD

**Issue by paales** [magento/magento2#6999]
### Preconditions

1. 2.1.2
2. Have configurables
### Steps to reproduce

1. After loading the configurable attributes it stores the result in the cache.
2. After loading the second time it tries to load it from the cache, but the `hasCacheData` method always returns false
### Expected result

1. Should use cache
### Actual result

1. Doesn't use cache

The two selected lines should be replaced:
https://github.com/magento/magento2/blob/1ade3b769a937f19682bbbe4e51b6c956147952f/app/code/Magento/ConfigurableProduct/Model/Product/Type/Configurable.php#L446-L447

With

``` PHP
$configurableAttributes = $configurableAttributes ? unserialize($configurableAttributes) : false;
if (is_object($configurableAttributes) && $configurableAttributes->count()) {
```

**Comment by thlassche** [comment]
@paales Your proposed check was not sufficient. I opened a pull request (#7376) for this.

**Comment by paales** [comment]
Hmm, I haven't experienced that it is still an array.. Maybe Magento could write some tests to check this ;)

**Comment by magento-team** [comment]
Internal ticket to track issue progress: MAGETWO-69587

**Comment by thlassche** [comment]
@magento-team It's already fixed in a merged PR to develop

**Comment by korostii** [comment]
Never mind that, @magento-team's comment seems to be automatic. The bot is probably walking through all the internal ticket references it has, no matter the status.

**Comment by magento-engcom-team** [comment]
@paales the issue already fixed in 2.1.10 and will be available soon
